transport/src/main/java/io/netty/channel/ManualIoEventLoop.java

```diff
@@ -0,0 +1,489 @@
+/*
+ * Copyright 2025 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel;
+
+import io.netty.util.concurrent.AbstractScheduledEventExecutor;
+import io.netty.util.concurrent.DefaultPromise;
+import io.netty.util.concurrent.Future;
+import io.netty.util.concurrent.GlobalEventExecutor;
+import io.netty.util.concurrent.Promise;
+import io.netty.util.internal.ObjectUtil;
+import io.netty.util.internal.PlatformDependent;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.Queue;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * {@link IoEventLoop} implementation that is owned by the user and so needs to be driven by the user manually with the
+ * given {@link Thread}. This means that the user is responsible to call either {@link #runNow()} or {@link #run(long)}
+ * to execute IO and tasks that were submitted to this {@link IoEventLoop}.
+ * <p>
+ * This is for <strong>advanced use-cases only</strong>, where the user wants to own the {@link Thread} that drives the
+ * {@link IoEventLoop} to also do other work. Care must be taken that the {@link #runNow() or
+ * {@link #waitAndRun()}} methods are called in a timely fashion.
+ */
+public final class ManualIoEventLoop extends AbstractScheduledEventExecutor implements IoEventLoop {
```

Comment:
Not sure about been final; 
Let's say (you know why) we want to have our own tasks queue (not the non blocking ones owned by this event loop) and we need to process and submit them (override the Executor::execute method) eventually calling Io handler::wakeup, we need to both have access to the handler and overrides the execute method.

I will write an example to show what I mean :)

Reply:
Yes please write an example